### PR TITLE
perf(webpack.config): enable caching of both sourcemaps and modules

### PR DESCRIPTION
**Pull request by jpumford**

### CODE REVIEW

**Comment by jpumford** [comment]
@sldavidson

File: other/webpack.config.es6.js [hunk 1]
```diff
@@ -40,7 +40,6 @@ const pkg = require('../package.json');
 const bannerText = fs.readFileSync(path.resolve(__dirname, '../BANNER.txt')).toString();
 
 const config = {
-  cache: false,
```
Comment (sldavidson) [inline other/webpack.config.es6.js]: When does the cache bust? We wouldn't run into issues with changes not showing up, right?

File: other/webpack.config.es6.js [hunk 2]
```diff
@@ -185,7 +184,7 @@ const config = {
     formatter: require('eslint-friendly-formatter'),
   },
 
-  devtool: 'source-map',
+  devtool: 'cheap-module-eval-source-map',
```
Comment (andrewjensen) [inline other/webpack.config.es6.js]: I think we only want this setting for `ON_DEV`. (And maybe `ON_TEST`.)

I tried running `npm run build` with these changes and it looks like it's actually putting all the prod code in `eval()` statements. It's also including the source mappings inline.

The downside to keeping `source-map` for production is that the one-time build takes quite a bit longer, but the better performance and security of the code is worth the wait.

Comment (jpumford) [inline other/webpack.config.es6.js]: Yeah, that's a good catch. Basically we need a ternary operator here. I misread one of the tables on webpack's "stellar" documentation to mean that this option was ignored on prod builds, but I never realized that webpack actually doesn't know what a prod build is.

I'll send that fix in tomorrow.

**Comment by jpumford** [comment]
@sldavidson it won't cache on any build except `npm start` incremental builds. It speeds up development by an order of magnitude 
Haha, there's no reply button to your comment.

**Comment by jpumford** [comment]
@andrewjensen How do you like that fix?

**Pull request merged**